Doors

BenoitRoux0 opened the pull request.

BenoitRoux0 committed:
"Added interactive doors feature in the game"

This commit adds an interactive doors feature in the game. In the game, we can now interact with the doors that can open and close using the 'E' key from the keyboard. The interaction with the doors is defined in the player input and player interact functions. New files for handling doors (doors_bonus.h, create_doors_map_bonus.c, destroy_doors_map_bonus.c) were added. It also added necessary checks in functions like ray_collides_bonus.c and update_player_bonus.c to handle player navigation through the doors. This interaction increases the complexity and interactivity of the game.

BenoitRoux0 committed:
"Improved collision detection and added door functionality"

This commit includes alterations to collision detection within the 'update_player' function. It introduces a finer detection mechanism by calculating vector positions between players and their surroundings to prevent potential collision anomalies. Also, a 'sign' function was added to help determine direction. This commit also modifies rendering operations in the 'select_tex' function to support the new door feature. A new method 'get_to_check' is introduced to streamline position calculation within this function. Lastly, the functionality to interact with doors has been introduced to the 'input_player_js' file, effectively enhancing player navigation and overall game interactivity.

BenoitRoux0 committed:
Correct use of minus sign for sign function

Fixed the conditional statement that checks the sign of a number. Previously, it was written incorrectly and it could give wrong results when the number is positive. Now, it correctly differentiates positive and negative numbers.

BenoitRoux0 committed:
Update sign function use in player update bonus

Improved the player bonus functionality by correcting the use of subtractive '-' sign with the 'sign' function. Previously, this would yield incorrect results for positive numbers unbalancing the gameplay, however, with the addition of the '+' sign, this function now appropriately differentiates between positive and negative numbers.

gd-harco commented on bonus/srcs/main_bonus.c:
```diff
@@ -22,28 +23,35 @@ int	main(int argc, char **argv)
 	data.mlx = mlx_init();
 	if (!data.mlx)
-		return (1);
+		return (on_destroy(&data), 1);
```
pourquoi ne pas juste appeler on_destroy, pas la peine de return ici si ?

BenoitRoux0 replied:
C'est vrai, les returns étaient déjà là et j'ai ajouter les on_destroy pour éviter tous leaks

gd-harco commented on bonus/srcs/main_bonus.c:
```diff
@@ -22,28 +23,35 @@ int	main(int argc, char **argv)
 	data.win = mlx_new_window(data.mlx, WIN_WIDTH, WIN_HEIGHT, "cub3D");
 	if (!data.win)
-		return (2);
+		return (on_destroy(&data), 2);
```
same

gd-harco commented on bonus/srcs/main_bonus.c:
```diff
@@ -22,28 +23,35 @@ int	main(int argc, char **argv)
 	data.img = init_img(WIN_WIDTH, WIN_HEIGHT);
 	if (!data.img)
-		return (3);
+		return (on_destroy(&data), 3);
```
same

BenoitRoux0 committed:
Adjust game error handling in main_bonus.c

Modified the way game handles errors to improve system stability. Previously, if an error occurred during initialization (e.g. unable to load a part of the game), the program would immediately terminate causing unexpected crashes. In the new implementation, the failing function, 'on_destroy()', is simply called without returning, allowing the game to clean up resources safely and proceed with the next steps of initialization.

gd-harco reviewed (approved):
Let's go
